### PR TITLE
[FEAT] RequestManager (emberjs/rfcs#860)

### PR DESCRIPTION
**Pull request by runspired**
Implements @ember-data/request Package from https://github.com/emberjs/rfcs/pull/860

progress towards finalizing #6166 
Required for #8086 

TODO

- [x] implement RequestManager
- [x] implement `Fetch` handler
- [x] add RequestManager tests
- [x] update docs and code import paths to reflect named import/export
- [x] re-review new READMEs
- [x] add API docs
- [x] update RFC with implementation notes and final interfaces

TODO (follow up)

- add experimental typescript definitions for RequestManager / Handler / Fetch handler
- makes `store.notifications` public to give access to the notification manager
- add `cache.peek` to give access to full resource data (emberjs/rfcs#854) 
- add support for `RequestManager` to the store
- add LegacyHandler

### CODE REVIEW

File: packages/request/README.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+```ts
+import Store from '@ember-data/store';
+import RequestManager from '@ember-data/request';
+
+export default class extends Store {
+  requestManager = new RequestManager();
+
+  constructor(args) {
+    super(args);
+    this.requestManager.use([LegacyHandler]);
```
Comment (sly7-7) [inline packages/request/README.md:355]: I just can't see where this `LegacyHandler` is coming from

Comment (runspired) [reply packages/request/README.md:355]: Oh I see, the import didn't make it into the ReadMe. It's in the rfc iirc, will add here too.

File: packages/store/README.md [hunk 2]
```diff
@@ -1,31 +1,180 @@
...
+```js
+import Store from '@ember-data/store';
+import Adapter from '@ember-data/adapter/json-api';
+import { getOwner, setOwner } from '@ember/application';
+
+class extends Store {
+  #adapter = null;
+
+  adapterFor() {
+    let adapter = thsi.#adapter;
```
Comment (sly7-7) [inline packages/store/README.md:123]: *this

File: packages/request/README.md (hunk 1 above)
Comment (sly7-7) [reply packages/request/README.md:355]: Sorry for not looking at the RFC, indeed I just see it there

**Pull request merged**